apps/pipelines/nodes/base.py

```diff
@@ -45,6 +45,10 @@ def add_temp_state_messages(left: dict, right: dict):
     return output
 
 
+def merge_dicts(left: dict, right: dict):
+    return {**left, **right}
```

Comment:
This is fine for dicts with simple values but if the value is a list, set or dict it won't merge them. I think you need a recursive merge strategy here.